src/main/java/org/spongepowered/api/event/entity/projectile/LaunchProjectileEvent.java

```diff
@@ -24,12 +24,31 @@
  */
 package org.spongepowered.api.event.entity.projectile;
 
+import org.spongepowered.api.entity.Entity;
 import org.spongepowered.api.entity.projectile.Projectile;
-import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.entity.AffectEntityEvent;
+import org.spongepowered.api.event.entity.ConstructEntityEvent;
+import org.spongepowered.api.event.entity.SpawnEntityEvent;
+import org.spongepowered.api.util.annotation.eventgen.PropertySettings;
+
+import java.util.List;
 
 /**
- * Called when a {@link Projectile} is launched.
+ * Called when a {@link Projectile} is launched. This event is always followed by a {@link SpawnEntityEvent}
```

Comment:
Why would this be followed by a spawn entity event and not just have this extend spawn entity event?

Reply:
I did this because in the Minecraft code, the entity is first created, then is launched, and is finally spawned in the world.